src/vaults/yields.utils.ts

```diff
@@ -463,3 +432,153 @@ function constructGraphVaultData(
     };
   });
 }
+
+export async function estimateVaultPerformance(
+  chain: Chain,
+  vault: VaultDefinitionModel,
+  data: VaultHarvestData[],
+): Promise<YieldSource[]> {
+  if (data.length === 0) {
+    return [];
+  }
+
+  const sdk = await chain.getSdk();
+  const recentHarvests = data.sort((a, b) => b.timestamp - a.timestamp);
+
+  /**
+   * ON 8/15 AN INCORREC PROCESSING OF A BADGER REWARDS PROCESSOR OCCURED.
+   * AS A RESULT, THE EVENTS INCLUDED IN TREE DISTRIBUTIONS DID NOT MAKE IT
+   * TO THE GRAPH, OR ANY SOURCE OF ON CHAIN DATA CURRENTLY SUPPORTED BY
+   * THE CURRENT YIELD SYSTEM.
+   *
+   * https://etherscan.io/tx/0x1e3e7c71012d36e936b768a37e9784125a00f205a22bd808f045968a506bb1ce#eventlog
+   *
+   * THIS TRANSACTION CONTAINS THE SINGLE BADGER TREE DISTRIBUTION WE ARE
+   * ALLOCATING TO GRAVI_AURA VAULT AS A MISSED - AND NOT CAPTURED SOURCE.
+   *
+   * THIS CODE SHOULD BE REMOVED BY 08/29.
+   */
+  if (vault.address === TOKENS.GRAVI_AURA) {
+    const targetBlock = 15344809;
+    const block = await sdk.provider.getBlock(targetBlock);
+    const targetedInsertion = recentHarvests[0].treeDistributions;
+    targetedInsertion.push({
+      timestamp: block.timestamp,
+      block: targetBlock,
+      token: TOKENS.BADGER,
+      amount: BigNumber.from('1928771715566995688546'),
+    });
+  }
+
+  const allHarvests = recentHarvests.flatMap((h) => h.harvests.map((h) => ({ ...h, type: HarvestType.Harvest })));
+  const allDistributions = recentHarvests.flatMap((h) =>
+    h.treeDistributions.map((d) => ({ ...d, type: HarvestType.TreeDistribution })),
+  );
+  const allEvents = allHarvests
+    .concat(allDistributions)
+    .filter((e) => BigNumber.from(e.amount).gt(ethers.constants.Zero))
+    .sort((a, b) => b.timestamp - a.timestamp);
+
+  let harvestReport = `${vault.name} Harvest Report\n`;
+
+  let totalHarvested = 0;
+  let totalAccumulated = 0;
+  let totalVaultTokens = 0;
+  let totalVaultPrincipal = 0;
+
+  console.log(`Evaluating ${allEvents.length} events for ${vault.name}`);
+  for (const event of allEvents) {
```

Comment:
parse through all events to capture and average capital yielding the given captured rewards,
here is where we would persist this input data to display on a 'yield inputs' page